dev-tools/smoke_test_rc.py

```diff
@@ -165,6 +169,11 @@ def download_and_verify(version, hash, files, base_url, plugins=DEFAULT_PLUGINS)
   finally:
     shutil.rmtree(tmp_dir)
 
+def get_host_from_ports_file(es_dir):
+  http_port = read_fully(os.path.join(es_dir, 'logs/http.ports')).splitlines()[0]
```

Comment:
I think the variable name `http_port` is misleading. In RFC3986 this is called [authority](http://tools.ietf.org/html/rfc3986#section-3.2) but I think `host_port` or something along those lines is also fine. Btw, I checked out of interest and all your logic works also fine for IPv6 addresses.